extensions/vercast/src/vercel.ts

```diff
@@ -36,52 +38,84 @@ export async function fetchUsername(): Promise<string> {
     showToast(ToastStyle.Failure, 'Failed to fetch username')
     throw new Error('Failed to fetch username')
   }
-  return Promise.resolve('')
+}
+
+export async function fetchTeams(
+  ignoredTeamIDs: string[]
+): Promise<Record<string, string>> {
+  const response = await fetch(apiURL + 'v1/teams', {
+    method: 'get',
+    headers: headers,
+  })
+  const json = await response.json()
+  const teams: Record<string, string> = {}
+  for (const team of json.teams) {
+    teams[team.id] = team.slug
+  }
+  if (ignoredTeamIDs.length > 0) {
+    for (const ignoredTeamID of ignoredTeamIDs) {
+      if (teams[ignoredTeamID]) {
+        delete teams[ignoredTeamID]
+      }
+    }
+  }
+  return teams
 }
 
 export async function fetchDeployments(
-  username: string
+  username: string,
+  teams: Record<string, string>
 ): Promise<Deployment[]> {
   dayjs.extend(relativeTime)
-
   try {
-    const response = await fetch(apiURL + 'v8/projects', {
-      method: 'get',
-      headers: headers,
-    })
-    const json = await response.json()
     const deployments: Deployment[] = []
-    for (const project of json.projects) {
-      for (const deployment of project.latestDeployments) {
-        if (deployment.creator.username === username) {
-          let state: DeploymentState
-          switch (deployment.readyState.toUpperCase()) {
-            case 'READY':
-              state = DeploymentState.ready
-              break
-            case 'BUILDING':
-            case 'QUEUED':
-              state = DeploymentState.deploying
-              break
-            default:
-              state = DeploymentState.failed
-              break
+    const teamIDs = Object.keys(teams)
+    for (let i = -1; i < teamIDs.length; i++) {
+      console.log(teams[teamIDs[i]])
+      const response = await fetch(
+        apiURL + `v8/projects${i === -1 ? '' : '?teamId=' + teams[teamIDs[i]]}`,
+        {
+          method: 'get',
+          headers: headers,
+        }
+      )
+      const json = await response.json()
+      for (const project of json.projects) {
+        for (const deployment of project.latestDeployments) {
+          if (deployment.creator.username === username) {
```

Comment:
Could that be the reason why I'm not using my teams' deployments? The list shows up empty for me – I guess that's because I'm not the creator of those deployments?